src/datadog/http_client.h

```diff
@@ -46,11 +47,11 @@ class HTTPClient {
   // preparing the request, return an `Error`.
   virtual Expected<void> post(const URL& url, HeadersSetter set_headers,
                               std::string body, ResponseHandler on_response,
-                              ErrorHandler on_error) = 0;
+                              ErrorHandler on_error, Timeout timeout) = 0;
 
   // Wait until there are no more outstanding requests, or until the specified
   // `deadline`.
-  virtual void drain() = 0;
+  virtual void drain(std::chrono::steady_clock::time_point deadline) = 0;
```

Comment:
I would prefer `drain(Timeout timeout)` instead but that requires passing the clock in the ctor and changing too many part of the codebase. Good side effect would be `test_curl` would use a fake clock.